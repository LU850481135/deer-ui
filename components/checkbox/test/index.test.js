import React from 'react';
import { render, mount,shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Checkbox from '../index';
const CheckboxGroup = Checkbox.CheckboxGroup

describe('Checkbox', () => {
  it('renders normal Checkbox', () => {
    const wrapper = render(
        <Checkbox value="Checkbox">
        复选框
      </Checkbox>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders disabled Checkbox', () => {
    const wrapper = render(
        <Checkbox value="Checkbox" disabled>
        复选框
      </Checkbox>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders indeterminate Checkbox', () => {
    const wrapper = render(
        <Checkbox value="Checkbox" indeterminate>
        复选框
      </Checkbox>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders checked Checkbox', () => {
    const wrapper = render(
        <Checkbox
        value="php"
        checked
      >
        Php
      </Checkbox>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should can trigger change event", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Checkbox onChange={onChange}  value="Checkbox">复选框</Checkbox>);
    wrapper.find("input").simulate("change");
    expect(onChange).toHaveBeenCalled();
  });

  it('renders CheckboxGroup correctly', () => {
    const plainOptions = ["Php", "Java", "Js"];
    const wrapper = render(
        <CheckboxGroup options={plainOptions} value={["php","java", "js"]}/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });


  it('renders defaultChecked Checkbox correctly', () => {
    const wrapper = render(
        <Checkbox
        value="js"
        defaultChecked
        onChange={e => {
          console.log(e.target.checked);
        }}
      >
        Javascript
      </Checkbox>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when checking in CheckboxGroup', () => {
    const onChange = jest.fn();
    const wrapper = mount(
        <CheckboxGroup
          options={["1", "2", "3"]}
          value={["1", "2", "3"]}
          onChange={onChange}
        />
    );

    wrapper.find('input').first(0).simulate('change');

    expect(onChange).toHaveBeenCalled();
  });
});
