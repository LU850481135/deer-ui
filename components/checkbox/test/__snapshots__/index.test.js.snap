// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders CheckboxGroup correctly 1`] = `
<div
  class="checkboxGroup"
>
  <div
    class="deer-ui-checkbox"
  >
    <label
      class="deer-ui-checkbox-wrapper"
    >
      <span
        class="deer-ui-checkbox-wrapper-content"
      >
        <input
          class="deer-ui-checkbox-checkbox"
          type="checkbox"
          value="php"
        />
        <span
          class="deer-ui-checkbox-inner"
        />
        <span>
          Php
        </span>
      </span>
    </label>
  </div>
  <div
    class="deer-ui-checkbox"
  >
    <label
      class="deer-ui-checkbox-wrapper"
    >
      <span
        class="deer-ui-checkbox-wrapper-content"
      >
        <input
          class="deer-ui-checkbox-checkbox"
          type="checkbox"
          value="java"
        />
        <span
          class="deer-ui-checkbox-inner"
        />
        <span>
          Java
        </span>
      </span>
    </label>
  </div>
  <div
    class="deer-ui-checkbox"
  >
    <label
      class="deer-ui-checkbox-wrapper"
    >
      <span
        class="deer-ui-checkbox-wrapper-content"
      >
        <input
          class="deer-ui-checkbox-checkbox"
          type="checkbox"
          value="js"
        />
        <span
          class="deer-ui-checkbox-inner"
        />
        <span>
          Js
        </span>
      </span>
    </label>
  </div>
</div>
`;

exports[`Checkbox renders checked Checkbox 1`] = `
<div
  class="deer-ui-checkbox"
>
  <label
    checked=""
    class="deer-ui-checkbox-wrapper"
  >
    <span
      class="deer-ui-checkbox-wrapper-content deer-ui-checkbox-wrapper-content-checked"
    >
      <input
        checked=""
        class="deer-ui-checkbox-checkbox"
        type="checkbox"
        value="php"
      />
      <span
        class="deer-ui-checkbox-inner"
      />
      <span>
        Php
      </span>
    </span>
  </label>
</div>
`;

exports[`Checkbox renders defaultChecked Checkbox correctly 1`] = `
<div
  class="deer-ui-checkbox"
>
  <label
    class="deer-ui-checkbox-wrapper"
  >
    <span
      class="deer-ui-checkbox-wrapper-content deer-ui-checkbox-wrapper-content-checked"
    >
      <input
        checked=""
        class="deer-ui-checkbox-checkbox"
        type="checkbox"
        value="js"
      />
      <span
        class="deer-ui-checkbox-inner"
      />
      <span>
        Javascript
      </span>
    </span>
  </label>
</div>
`;

exports[`Checkbox renders disabled Checkbox 1`] = `
<div
  class="deer-ui-checkbox"
>
  <label
    class="deer-ui-checkbox-wrapper deer-ui-checkbox-wrapper-disabled"
  >
    <span
      class="deer-ui-checkbox-wrapper-content"
    >
      <input
        class="deer-ui-checkbox-checkbox"
        disabled=""
        type="checkbox"
        value="Checkbox"
      />
      <span
        class="deer-ui-checkbox-inner"
      />
      <span>
        复选框
      </span>
    </span>
  </label>
</div>
`;

exports[`Checkbox renders indeterminate Checkbox 1`] = `
<div
  class="deer-ui-checkbox"
>
  <label
    class="deer-ui-checkbox-wrapper"
  >
    <span
      class="deer-ui-checkbox-wrapper-content"
    >
      <input
        class="deer-ui-checkbox-checkbox"
        type="checkbox"
        value="Checkbox"
      />
      <span
        class="deer-ui-checkbox-inner"
      />
      <span>
        复选框
      </span>
    </span>
  </label>
</div>
`;

exports[`Checkbox renders normal Checkbox 1`] = `
<div
  class="deer-ui-checkbox"
>
  <label
    class="deer-ui-checkbox-wrapper"
  >
    <span
      class="deer-ui-checkbox-wrapper-content"
    >
      <input
        class="deer-ui-checkbox-checkbox"
        type="checkbox"
        value="Checkbox"
      />
      <span
        class="deer-ui-checkbox-inner"
      />
      <span>
        复选框
      </span>
    </span>
  </label>
</div>
`;
